main.go

```diff
@@ -110,7 +116,9 @@ func sendmail(cfg SendConfig, ID int) {
 	}
 	// Else update the send status
 	mu.Lock()
-	MailSentMap[ID] = 1
+	//MailSentMap[ID] = 1
+	db.Model(&user).Where("id = ?", ID).Update("last_send_email_time", int(time.Now().Unix()))
```

Comment:
Do not write model manipulation in the service routine
Write a new function in model.go